homeassistant/components/device_tracker/owntracks.py

```diff
@@ -56,15 +63,63 @@ def setup_scanner(hass, config, see):
     max_gps_accuracy = config.get(CONF_MAX_GPS_ACCURACY)
     waypoint_import = config.get(CONF_WAYPOINT_IMPORT)
     waypoint_whitelist = config.get(CONF_WAYPOINT_WHITELIST)
+    secret = config.get(CONF_SECRET)
 
-    def validate_payload(payload, data_type):
+    def decrypt_payload(topic, ciphertext):
+        """Decrypt encrypted payload."""
+        try:
```

Comment:
To make it more testable, add a method `get_libnacl()` that returns `KEYBYTES` and `SecretBox` from `libnacl` but can also be mocked in tests.